### PR TITLE
docs(cn): translate src/content/migrate/3.md

### PR DESCRIPTION
**Pull request by miazhuce**
_describe your changes..._
从 v1 升级到 v2 或 v3 @dear-lizhihua
- [ ] Read and sign the [CLA][1]. PRs that haven't signed it won't be accepted.
- [ ] Make sure your PR complies with the [writer's guide][2].
- [ ] Review the diff carefully as sometimes this can reveal issues.
- [ ] Do not abandon your Pull Request: [Stale Pull Requests][3].
- __Remove these instructions from your PR as they are for your eyes only.__

[1]: https://cla.js.foundation/webpack/webpack.js.org
[2]: https://webpack.js.org/contribute/writers-guide/
[3]: https://webpack.js.org/contribute/#pull-requests

### CODE REVIEW

File: src/content/migrate/3.md [hunk 1]
```diff
@@ -18,14 +18,14 @@ contributors:
   - byzyk
 ---
 
-The following sections describe the major changes from webpack 1 to 2.
+以下各节描述从 webpack 1 到 webpack 2 的重大变化。
 
-T> Note that there were far fewer changes between 2 and 3, so that migration shouldn't be too bad. If you are running into issues, please see [the changelog](https://github.com/webpack/webpack/releases) for details.
+T> 注意：webpack 从 1 到 2 的变化，比从 2 到 3 要少很多，所以版本迁移起来难度应该不大。如果你遇到了问题，请查看 [更新日志](https://github.com/webpack/webpack/releases) 以了解更多细节。
```
Comment (lcxfs1991) [inline src/content/migrate/3.md]: 请保证中英文的行数都保持一致

File: src/content/migrate/3.md [hunk 2]
```diff
@@ -40,19 +40,19 @@ These options were replaced by a single option `resolve.modules`. See [resolving
...
 ## `resolve.*`
 
-Several APIs were changed here. Not listed in detail as it's not commonly used. See [resolving](/configuration/resolve) for details.
+这里更改了几个 API 。由于不常用，不在这里详细列出。更多用法请查看 [解析](/configuration/resolve) 。
 
 
-## `module.loaders` is now `module.rules`
+## `module.loaders` 升级为 `module.rules`
+
+旧的 loader 配置被更强大的 rules 系统取代，后者允许配置 loader 以及其他更多选项。 为了兼容旧版，`module.loaders` 语法仍然有效，旧的属性名依然可以被解析。 新的命名约定更易于理解，并且是升级配置使用 `module.rules` 的好理由。
```
Comment (lcxfs1991) [inline src/content/migrate/3.md]: 行数请保持一致了

**Comment by lcxfs1991** [comment]
@miazhuce 你好，麻烦将丢掉的格式都补齐哈。尤其是丢掉的 - 符号

File: src/content/migrate/3.md [hunk 3]
```diff
@@ -380,15 +380,15 @@ module.exports = {
...
   ts: { transpileOnly: false }
 };
 ```
 
 
-### What are `options`?
+### 什么是 `options` ？
 
-Good question. Well, strictly speaking it's 2 possible things; both ways to configure a webpack loader. Classically `options` was called `query` and was a string which could be appended to the name of the loader. Much like a query string but actually with [greater powers](https://github.com/webpack/loader-utils#parsequery):
+好问题。严格来说，有两种办法，都可以用来配置 webpack 的 loader 。典型的 `options` 被称为 `query`，是一个可以被添加到 loader 名之后的字符串。它比较像一个 query string，但是实际上有[更强大的能力](https://github.com/webpack/loader-utils#parsequery):
```
Comment (lcxfs1991) [inline src/content/migrate/3.md]: query string -> 查询字符串(query string)

File: src/content/migrate/3.md [hunk 4]
```diff
@@ -561,74 +561,74 @@ module.exports = function() {
...
   //...
   module: {
     rules: [
       {
-        resource: /filename/, // matches "/path/filename.js"
-        resourceQuery: /^\?querystring$/, // matches "?querystring"
-        issuer: /filename/, // matches "/path/something.js" if requested from "/path/filename.js"
+        resource: /filename/, // 匹配 "/path/filename.js"
+        resourceQuery: /^\?querystring$/, // 匹配 "?querystring"
+        issuer: /filename/, // 果请求 "/path/filename.js" 则匹配 "/path/something.js"
```
Comment (lcxfs1991) [inline src/content/migrate/3.md]: 如果

**Review by lcxfs1991 — CHANGES_REQUESTED** [review]

File: src/content/migrate/3.md [hunk 5]
```diff
@@ -130,21 +130,21 @@ It is not possible anymore to omit the `-loader` extension when referencing load
...
 ```
 
-See [#2986](https://github.com/webpack/webpack/issues/2986) for the reason behind this change.
+了解这一改变背后的原因，请查看 [#2986](https://github.com/webpack/webpack/issues/2986) 。
 
 
-## `json-loader` is not required anymore
+## `json-loader` 不再需要手动添加
+
+如果没有为 JSON 文件配置 loader，webpack 将自动尝试通过 [`json-loader`](https://github.com/webpack-contrib/json-loader) 加载 JSON 文件。
```
Comment (QC-L) [inline src/content/migrate/3.md]: 多余的空行请删掉

File: src/content/migrate/3.md (hunk 1 above)
Comment (miazhuce) [reply src/content/migrate/3.md]: 修改完成提交

File: src/content/migrate/3.md (hunk 2 above)
Comment (miazhuce) [reply src/content/migrate/3.md]: 修改完成提交

File: src/content/migrate/3.md (hunk 3 above)
Comment (miazhuce) [reply src/content/migrate/3.md]: 修改完成提交

File: src/content/migrate/3.md (hunk 4 above)
Comment (miazhuce) [reply src/content/migrate/3.md]: 修改完成提交

File: src/content/migrate/3.md (hunk 5 above)
Comment (miazhuce) [reply src/content/migrate/3.md:147]: 额 跟master分支的3.md的一样
image

Comment (miazhuce) [reply src/content/migrate/3.md]: done

File: src/content/migrate/3.md [hunk 6]
```diff
@@ -87,11 +87,10 @@ The new naming conventions are easier to understand and are a good reason to upg
 ```
 
 
-## Chaining loaders
+## 链式 loaders
+
+就像在 webpack 1 中，loader 可以链式调用，上一个 loader 的输出被作为输入传给下一个 loader。 使用 [rule.use](/configuration/module/#ruleuse) 配置选项，`use` 可以设置为一个 loader 数组。 在 webpack 1 中，loader 通常被用 `!` 连写。这一写法在 webpack 2 中只在使用旧的选项 `module.loaders` 时才有效。
```
Comment (QC-L) [inline src/content/migrate/3.md]:
```suggestion
```

File: src/content/migrate/3.md [hunk 7]
```diff
@@ -87,11 +87,9 @@ The new naming conventions are easier to understand and are a good reason to upg
 ```
 
 
-## Chaining loaders
+## 链式 loaders
 
-Like in webpack 1, loaders can be chained to pass results from loader to loader. Using the [rule.use](/configuration/module/#ruleuse)
- configuration option, `use` can be set to an array of loaders.
-In webpack 1, loaders were commonly chained with `!`. This style is only supported using the legacy option `module.loaders`.
+就像在 webpack 1 中，loader 可以链式调用，上一个 loader 的输出被作为输入传给下一个 loader。 使用 [rule.use](/configuration/module/#ruleuse) 配置选项，`use` 可以设置为一个 loader 数组。 在 webpack 1 中，loader 通常被用 `!` 连写。这一写法在 webpack 2 中只在使用旧的选项 `module.loaders` 时才有效。
```
Comment (QC-L) [inline src/content/migrate/3.md]:
```suggestion
就像在 webpack 1 中，loader 可以链式调用，上一个 loader 的输出被作为输入传给下一个 loader。
使用 [rule.use](/configuration/module/#ruleuse) 配置选项，`use` 可以设置为一个 loader 数组。
在 webpack 1 中，loader 通常被用 `!` 连写。这一写法在 webpack 2 中只在使用旧的选项 `module.loaders` 时才有效。
```

**Review by QC-L — APPROVED** [review]

**Pull request merged**